### PR TITLE
Hardening code by adding additonal exception handling structures

### PR DESCRIPTION
**Pull request by DukeDavis12**
Hardening code by adding additional exception handling structures.
   Policy used: 
   - caller function handles all the exceptions thrown by
   the functions in their execution stack.
  - Exceptions in REST APIs are handled in their implementation itself.
 
Signed-off-by: Davis Benny <davis.benny@intel.com>

### CODE REVIEW

File: component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioContextListener.java [hunk 1]
```diff
@@ -92,14 +92,17 @@ public void contextInitialized(ServletContextEvent sce) {
...
+      String epidUrl = sc.getInitParameter(AioAppSettings.EPID_URL);
+      if (null != epidUrl) {
+        EpidUtils.setEpidOnlineUrl(epidUrl);
+      }
+    } catch (IllegalArgumentException e) {
       logger.info("EPID URL not set. Default URL will be used: "
           + EpidUtils.getEpidOnlineUrl().toString());
     }
 
+
```
Comment (DukeDavis12) [inline component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioContextListener.java:105]: Remove whitespace

File: component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioFileDownloadServlet.java [hunk 2]
```diff
@@ -7,18 +7,23 @@
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.nio.channels.ScatteringByteChannel;
```
Comment (DukeDavis12) [inline component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioFileDownloadServlet.java:10]: Remove this import

File: component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioFileUploadServlet.java [hunk 3]
```diff
@@ -16,10 +16,14 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+
```
Comment (DukeDavis12) [inline component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioFileUploadServlet.java:19]: remove whitespace

File: component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/Device.java [hunk 4]
```diff
@@ -54,7 +54,7 @@
   private static final String PROPERTY_DI_URL = "fidoalliance.fdo.url.di";
   private static final String PROPERTY_RANDOMS = "fidoalliance.fdo.randoms";
   private static final String PROPERTY_SERVICE_INFO_MTU =
-      "fidoalliance.fdo.device.service.info.mtu";
+          "fidoalliance.fdo.device.service.info.mtu";
```
Comment (DukeDavis12) [inline component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/Device.java:57]: remove whitespace

Comment (DukeDavis12) [reply component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/Device.java:57]: ++ Entire file

File: component-samples/manufacturer/src/main/java/org/fidoalliance/fdo/sample/ManufacturerApp.java [hunk 5]
```diff
@@ -183,18 +183,22 @@ public static void main(String[] args) {
...
+                ManufacturerConfigLoader.loadConfig(ManufacturerAppSettings.SSL_KEYSTORE_PASSWORD);
+
+        httpsConnector.setProperty("keystorePass", keystorePass);
+        httpsConnector.setProperty("keystoreFile", keyStoreFile.toFile().getAbsolutePath());
+        httpsConnector.setProperty("clientAuth", "false");
+        httpsConnector.setProperty("sslProtocol", "TLS");
+        httpsConnector.setProperty("SSLEnabled", "true");
+        service.addConnector(httpsConnector);
+      } catch (Exception e) {
+        logger.error("Error while loading keystore file");
```
Comment (DukeDavis12) [inline component-samples/manufacturer/src/main/java/org/fidoalliance/fdo/sample/ManufacturerApp.java:200]: Change error log.

File: http-api-samples/http-api-di-sample/src/main/java/org/fidoalliance/fdo/api/DiApiServlet.java [hunk 6]
```diff
@@ -60,52 +60,55 @@ protected Composite queryVoucher(DataSource dataSource, String serialNo) {
...
+      String serialNo = uri.substring(uri.lastIndexOf('/') + 1);
+
+      DataSource ds = (DataSource) getServletContext().getAttribute("datasource");
+      CryptoService cs = (CryptoService) getServletContext().getAttribute("cryptoservice");
+      CertificateResolver resolver =
+              (CertificateResolver) getServletContext().getAttribute("resolver");
+
+      Composite result = queryVoucher(ds, serialNo);
+      if (result.size() == 0) {
+        logger.warn("Request failed because of device with given serial was not found.");
```
Comment (DukeDavis12) [inline http-api-samples/http-api-di-sample/src/main/java/org/fidoalliance/fdo/api/DiApiServlet.java:74]: Change log statement -- Invalid serial number.

**Review by prateekcr — CHANGES_REQUESTED** [review]

File: util/http-dispatchers/http-server-dispatcher/src/main/java/org/fidoalliance/fdo/sample/ProtocolHandler.java [hunk 7]
```diff
@@ -90,56 +90,62 @@ public void run() {
...
-
     try {
-      Composite msg = getMessage(request);
-      DispatchResult result = dispatcher.dispatch(msg);
-      Composite reply = result.getReply();
-
-      if (reply.size() == 0) {
-        response.setContentLength(0);
-        response.setStatus(HttpServletResponse.SC_OK);
+      if (request.getContentType() == null
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply util/http-dispatchers/http-server-dispatcher/src/main/java/org/fidoalliance/fdo/sample/ProtocolHandler.java:94]: Removed.

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/cbor/Encoder.java [hunk 8]
```diff
@@ -76,12 +88,17 @@ public Encoder writeLong(Long val) throws IOException {
    * Encode a java Map as a CBOR map.
    */
   public <K, V> Encoder writeMap(Map<K, V> val) throws IOException {
-    startMap(val.size());
-    for (Map.Entry<K, V> e : val.entrySet()) {
-      writeObject(e.getKey());
-      writeObject(e.getValue());
+    try {
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/cbor/Encoder.java:91]: Removed.

File: http-api-samples/http-api-owner-sample/src/main/java/org/fidoalliance/fdo/api/OwnerCustomerServlet.java [hunk 9]
```diff
@@ -23,36 +23,43 @@
...
-    String id = req.getParameter("id");
-    String name = req.getParameter("name");
-    String contentType = req.getContentType();
-
-    if (null == id || null == name) {
-      logger.warn("Request failed because of invalid input.");
-      resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-      return;
-    }
+    try {
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply http-api-samples/http-api-owner-sample/src/main/java/org/fidoalliance/fdo/api/OwnerCustomerServlet.java:26]: Removed.

File: component-samples/rv/src/main/java/org/fidoalliance/fdo/sample/RvServerApp.java [hunk 10]
```diff
@@ -93,22 +93,25 @@ public static void main(String[] args) {
...
-      httpsConnector.setProperty("clientAuth", "false");
-      httpsConnector.setProperty("sslProtocol", "TLS");
-      httpsConnector.setProperty("SSLEnabled", "true");
-      service.addConnector(httpsConnector);
+      try {
+        httpsConnector.setPort(RV_HTTPS_PORT);
+        httpsConnector.setSecure(true);
+        httpsConnector.setScheme(RV_SCHEME);
+
+        Path keyStoreFile =
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/rv/src/main/java/org/fidoalliance/fdo/sample/RvServerApp.java:101]: I think it's better to cover the entire section in try-catch as any raised exception, prevents any half-baked httpsConnector object to be created and added to the service.

File: component-samples/rv/src/main/java/org/fidoalliance/fdo/sample/RvContextListener.java [hunk 11]
```diff
@@ -50,18 +50,24 @@ public void contextInitialized(ServletContextEvent sce) {
...
-      logger.warn("*** WARNING ***");
-      logger.warn("EPID Test mode enabled. This should NOT be enabled in production deployment.");
-    }
-    String epidUrl = sc.getInitParameter(RvAppSettings.EPID_URL);
-    if (null != epidUrl) {
-      EpidUtils.setEpidOnlineUrl(epidUrl);
-    } else {
-      logger.info("EPID URL not set. Default URL will be used: "
-              + EpidUtils.getEpidOnlineUrl().toString());
+    try {
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/rv/src/main/java/org/fidoalliance/fdo/sample/RvContextListener.java:53]: Removed,

File: component-samples/reseller/src/main/java/org/fidoalliance/fdo/sample/ResellerServerApp.java [hunk 12]
```diff
@@ -125,21 +125,25 @@ public static void main(String[] args) {
...
-      httpsConnector.setProperty("clientAuth", "false");
-      httpsConnector.setProperty("sslProtocol", "TLS");
-      httpsConnector.setProperty("SSLEnabled", "true");
-      service.addConnector(httpsConnector);
+      try {
+        httpsConnector.setPort(RESELLER_HTTPS_PORT);
+        httpsConnector.setSecure(true);
+        httpsConnector.setScheme(RESELLER_SCHEME);
+
+        Path keyStoreFile =
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/reseller/src/main/java/org/fidoalliance/fdo/sample/ResellerServerApp.java:133]: I think it's better to cover the entire section in try-catch as raised exception, prevents any half-baked httpsConnector object to be created and added to the service.

File: component-samples/manufacturer/src/main/java/org/fidoalliance/fdo/sample/ManufacturerContextListener.java [hunk 13]
```diff
@@ -189,26 +195,30 @@ protected void dispatching(Composite request) {
...
-    }
     try {
-      final String resellerKeysPem = Files.readString(Paths.get(
-              sc.getInitParameter(ManufacturerAppSettings.RESELLER_PUB_KEY_PATH)));
-      manager.addCustomer(ds, 2, "reseller", resellerKeysPem);
-      logger.info("Registered public keys for customer 'reseller'");
-    } catch (IOException e) {
-      logger.info("No default public keys found for customer 'reseller'");
+      //create tables
+      DiDbStorage db = new DiDbStorage(cs, ds, keyResolver, ods);
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/manufacturer/src/main/java/org/fidoalliance/fdo/sample/ManufacturerContextListener.java:200]: Removed.

File: component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioServerApp.java [hunk 14]
```diff
@@ -250,22 +251,25 @@ public static void main(String[] args) {
...
-      httpsConnector.setProperty("clientAuth", "false");
-      httpsConnector.setProperty("sslProtocol", "TLS");
-      httpsConnector.setProperty("SSLEnabled", "true");
-      service.addConnector(httpsConnector);
+      try {
+        httpsConnector.setPort(AIO_HTTPS_PORT);
+        httpsConnector.setSecure(true);
+        httpsConnector.setScheme(AIO_SCHEME);
+
+        Path keyStoreFile =
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioServerApp.java:259]: I think it's better to cover the entire section in try-catch as raised exception, prevents any half-baked httpsConnector object to be created and added to the service.

File: component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/DeviceSysModule.java [hunk 15]
```diff
@@ -80,34 +80,39 @@ public Composite getState() {
...
-      case FdoSys.KEY_EXEC:
-        if (isActive) {
-          exec(kvPair.getAsComposite(Const.SECOND_KEY));
-        } else {
-          logger.warn("fdo_sys module not active. Ignoring fdo_sys:exec.");
-        }
-        break;
-      default:
-        break;
+    try {
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/DeviceSysModule.java:83]: Done.

File: component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/Device.java [hunk 16]
```diff
@@ -586,10 +605,14 @@ void run() throws Exception {
   }
 
   void fail(Throwable cause) {
-    if (cause instanceof RuntimeException) {
-      throw (RuntimeException)cause;
-    } else {
-      throw new RuntimeException(cause);
+    try {
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/http-device-sample/src/main/java/org/fidoalliance/fdo/sample/Device.java:608]: Let's revisit this again.

File: protocol-samples/http-client-to2-sample/src/main/java/org/fidoalliance/fdo/sample/To2ClientApp.java [hunk 17]
```diff
@@ -69,28 +69,31 @@
   private void run(String[] args)
       throws NoSuchAlgorithmException, IOException, InterruptedException {
 
-    cryptoService = new CryptoService();
+    try {
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply protocol-samples/http-client-to2-sample/src/main/java/org/fidoalliance/fdo/sample/To2ClientApp.java:72]: Done.

File: protocol-samples/http-client-to1-sample/src/main/java/org/fidoalliance/fdo/sample/To1ClientApp.java [hunk 18]
```diff
@@ -53,28 +53,32 @@
...
-    signedTo1Blob = null;
-    for (String path : paths) {
-
-      try {
-        WebClient client = new WebClient(path, clientService.getHelloMessage(), dispatcher);
-        client.call();
-        if (signedTo1Blob != null) {
-          break;
+    try {
+      final MessageDispatcher dispatcher = getDispatcher();
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply protocol-samples/http-client-to1-sample/src/main/java/org/fidoalliance/fdo/sample/To1ClientApp.java:57]: Done.

File: component-samples/rv/src/main/java/org/fidoalliance/fdo/sample/RvServerApp.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/rv/src/main/java/org/fidoalliance/fdo/sample/RvServerApp.java:101]: Updated.

File: protocol-samples/http-client-to0-sample/src/main/java/org/fidoalliance/fdo/sample/To0ClientApp.java [hunk 19]
```diff
@@ -271,40 +271,44 @@ protected void failed(Exception e) {
...
+      Composite rvi = ovh.getAsComposite(Const.OVH_RENDEZVOUS_INFO);
 
 
-    // Protocol to0_to1_server (rv) only support http so we
-    // want to select the RV_DEV_ONLY filter here so the directive
-    // returned is http.
-    List<String> paths = RendezvousInfoDecoder.getHttpDirectives(
-        rvi,
-        Const.RV_DEV_ONLY);
+      // Protocol to0_to1_server (rv) only support http so we
```
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply protocol-samples/http-client-to0-sample/src/main/java/org/fidoalliance/fdo/sample/To0ClientApp.java:289]: Done.

File: component-samples/reseller/src/main/java/org/fidoalliance/fdo/sample/ResellerServerApp.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (DukeDavis12) [reply component-samples/reseller/src/main/java/org/fidoalliance/fdo/sample/ResellerServerApp.java:133]: Updated.

**Review by prateekcr — APPROVED** [review]

**Pull request closed without merging**